### ISSUE TITLE
Seldon MLServer only supports a single input tensor 

### ISSUE THREAD

**Issue by amnpandey** [SeldonIO/MLServer#342]
We are getting below error when we are trying KFS V2 request with multiple tensors:
```
InvalidArgument (3)
inference.GRPCInferenceService/ModelInfer: INVALID_ARGUMENT: SKLearnModel only supports a single input tensor (4 were received)
```

we have checked the implementations and it looks like it has been designed to work with only single tensor.
https://github.com/SeldonIO/MLServer/blob/56a90a45119ed5ab3cd7c99c68a109f5426828f6/runtimes/sklearn/mlserver_sklearn/sklearn.py#L47

Any thought(s) to make it work for multiple tensors. Thanks

**Comment by amnpandey** [comment]
closing this for now as we would like to investigate more on our side and then come back. Thanks

**Comment by joerunde** [comment]
@adriangonz Are you actively looking into this?

Not sure how you guys like to handle code snippets, but I put up a reproduction of this in https://github.com/SeldonIO/MLServer/pull/379. I'd like to try to contribute a fix, I think I just might need a bit of guidance on how to go about changing the input processing logic in the sklearn runtime without accidentally breaking any other use cases of the runtime.

Is there a set of test suites somewhere with some canonical cases of SKLearn models that should be serveable by the SKLearn runtime? Or do you all rely on input from other contributors and users to try to maintain compatibility with existing use cases?

Any guidance is appreciated!

**Comment by adriangonz** [comment]
Hey @joerunde , that'd be really welcome! Thanks a lot for jumping on this one. :+1: 

I've just realised that you've opened a draft PR with the changes, so we can keep the conversation going there.

**Comment by adriangonz** [comment]
Fixed by #379
